edgexfoundry/edgex-go
# build(snap): upgrade device-virtual  and app-service-config to 2.1

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

```
git clone https://github.com/MonicaisHer/edgex-go.git
cd edgex-go
git checkout snap-update-devicevirtual-asc-kuiper-redis
snapcraft
snap install edgexfoundry_2.0.1-dev.71_amd64.snap --dangerous
```
1. Validate that device-virtual 2.1 is running. 
```
snap set edgexfoundry device-virtual=on
snap logs edgexfoundry.device-virtual
```
Expect:
```
app=device-virtual source=message.go:51 msg="Starting device-virtual 2.1.0 "   
```
2. Validate that app-service-configurable is disabled and inactive by default. 
```
snap services edgexfoundry.app-service-configurable
```
Expect:
```
Service                                Startup   Current   Notes
edgexfoundry.app-service-configurable  disabled  inactive  -
```
3. Validate that app-service-configurable can be triggered by kuiper. 
```
snap set edgexfoundry kuiper=on
snap services edgexfoundry.app-service-configurable
```
Expect:
```
Service                                Startup  Current  Notes
edgexfoundry.app-service-configurable  enabled  active   -
```
4. Validate that app-service-configurable is running with the rules-engine profile. 
```
snap logs edgexfoundry.app-service-configurable
```
Expect:
```
level=INFO ts=2021-11-29T08:02:44.901337203Z app=app-rules-engine source=service.go:184 msg="app-rules-engine has Started"
```
5. validate that redis is runing.
```
snap logs edgexfoundry.redis
```
Expect:
```
Redis version=6.2.6, bits=64, commit=4930d19e, modified=0, pid=38151, just started
```
## New Dependency Instructions (If applicable)
